### DISCUSSION TITLE
Clients take a long time to connect to the server.

### DISCUSSION THREAD

**Discussion by Junglaverde** [ltsp/ltsp#707] [Q&A]
Hello, I have a server with ubuntu-mate, an i7 processor and 32gb of ram, and 14 computers with 4gb of ram, dual nic. Everything works very well. But they take a long time to connect to the server and start uploading the image.
Is there a way to make them connect faster?

**Comment by alkisg** [comment]
First, do an Epoptes LAN benchmark: https://epoptes.org/documentation/lan-benchmark/
Do you see any clients with bandwidth < 100Mbps?

Second, could you describe what part takes a long time? Is it only the "loading vmlinuz ... loading initrd ... " part, or all booting is slow?

**Comment by Junglaverde** [comment]
It's just late at first. The computers are waiting for the connection to the server. When they have already connected, the menu is fast and the image loads too. Then I check the LAN with epoptes and tell you something. Thank you very much.

**Comment by alkisg** [comment]
Try to get a screenshot from the point where they're waiting, so that we see what you're talking about. E.g. it might be your BIOS/UEFI firmware, or it might be iPXE. Or they might just be waiting on your second NIC, which you can disable from the BIOS if you're not using it.

A video would be best, but a screenshot should be OK too.

**Comment by Junglaverde** [comment]
Hello again, they are waiting here.

![IMG20](https://user-images.githubusercontent.com/81898780/165706085-cea52005-f9c6-4471-92e0-3bfa2233393a.jpg)

**Comment by Junglaverde** [comment]
The result of LAN benchmark epoptes is as follows.
![epoptes](https://user-images.githubusercontent.com/81898780/165706817-6b4b56e3-4896-4f06-a13f-bdbfa4fd9b2b.png)

**Comment by Junglaverde** [comment]
I have tested with a single computer and it reaches 900 Mbps in epoptes but the startup is just as slow.

**Comment by alkisg** [comment]
The screenshot that you show means that your DHCP server doesn't respond quickly.
On your LTSP server, type `ip a`. See the name of your Ethernet adapter. In the following command, substitute `enp2s0` with the name that you saw. What is the output of that command?

```shell
sudo time /usr/lib/klibc/bin/ipconfig -n enp2s0
```

**Comment by Junglaverde** [comment]
![ip_a](https://user-images.githubusercontent.com/81898780/165714544-318ee3e1-a601-46e0-9667-eb8754236fb2.png)

**Comment by alkisg** [comment]
You forgot the "lib" part in the path, it's `sudo time /usr/lib/klibc/bin/ipconfig -n enp0s25`.
The clients that have this issue, are they connected with enp1s0=192.168.67.1, or with enp0s25=172.24.224.4?
Whey they end up booting, what IP do they get, in the 192. range or in the 172. range?

**Comment by Junglaverde** [comment]
The clients are they connected with enp1s0=192.168.67.1
172.24.224.1 is ADSL

**Comment by alkisg** [comment]
OK. Now boot an LTSP client and open a terminal.
Run the same sudo time command on the LTSP client and paste the result.

**Comment by Junglaverde** [comment]
server
![server](https://user-images.githubusercontent.com/81898780/165717064-1cef1e54-691a-4d25-b326-71765c6f6bf8.png)

**Comment by Junglaverde** [comment]
client
![client](https://user-images.githubusercontent.com/81898780/165717142-5aba0632-44e1-4677-b022-6851ef460451.jpg)

**Comment by alkisg** [comment]
I don't see any issue with the DHCP server, it replies very fast there.

Are you sure that the 192.x and 172.x networks are not physically connected by mistake? They should be in completely separate switches with no connection between them.

Another test would be to write http://boot.ipxe.org/ipxe.usb or https://github.com/alkisg/liveusb in a USB stick (e.g. using the dd command) and boot an LTSP client with it; iPXE will be able to report networking issues much better than the internal PXE stack.

**Comment by Junglaverde** [comment]
Hello Alkisg, I'll give you information about the installation and a video of the usb
![configuracion](https://user-images.githubusercontent.com/81898780/165756653-94efa303-5245-430b-9084-b74e79f9f0f8.png)
![install](https://user-images.githubusercontent.com/81898780/165756666-5450db2e-ca1f-42d0-9846-50fbab28d141.png)

https://user-images.githubusercontent.com/81898780/165757982-40f21192-cc01-4680-a832-f9a2594d8d9d.mp4

**Comment by alkisg** [comment]
OK, so the DHCP request at boot is taking 6 seconds instead of 1 second.
I don't see any misconfiguration from the LTSP side.
You could talk to the iPXE people about this and see if they can guide you through network traffic packet inspection, to see if there are any dropped DHCPOFFERs or whatever else might be delaying things, but personally I don't have anything substantial to suggest.
You could also try [isc-dhcp](https://ltsp.org/advanced/isc-dhcp-server/) instead of dnsmasq, to see if it makes any difference, but personally I wouldn't bother for 5 seconds.

**Comment by Junglaverde** [comment]
Well, when I turn them all on at once it takes a long time. Since I have another server, I'm going to try another type of switch in case that's the problem.
Thank you very much for your help.
